core/src/main/java/org/pentaho/di/core/row/value/ValueMetaTimestamp.java

```diff
@@ -0,0 +1,709 @@
+      outputStream.writeBoolean( object == null );
+
+      if ( object != null ) {
+        switch ( storageType ) {
+          case STORAGE_TYPE_NORMAL:
+            // Handle Content -- only when not NULL
+            Timestamp timestamp = convertDateToTimestamp( (Date) object );
+
+            outputStream.writeLong( timestamp.getTime() );
+            outputStream.writeInt( timestamp.getNanos() );
+            break;
+
+          case STORAGE_TYPE_BINARY_STRING:
+            // Handle binary string content -- only when not NULL
+            // In this case, we opt not to convert anything at all for speed.
+            // That way, we can save on CPU power.
+            // Since the streams can be compressed, volume shouldn't be an issue
+            // at all.
+            //
+            writeBinaryString( outputStream, (byte[]) object );
+            break;
+
+          case STORAGE_TYPE_INDEXED:
+            writeInteger( outputStream, (Integer) object ); // just an index
+            break;
+
+          default:
+            throw new KettleFileException( toString() + " : Unknown storage type " + getStorageType() );
+        }
+      }
+    } catch ( ClassCastException e ) {
+      throw new RuntimeException( toString() + " : There was a data type error: the data type of "
+        + object.getClass().getName() + " object [" + object + "] does not correspond to value meta ["
+        + toStringMeta() + "]" );
+    } catch ( IOException e ) {
+      throw new KettleFileException( toString() + " : Unable to write value timestamp data to output stream", e );
+    } catch ( KettleValueException e ) {
+      throw new RuntimeException( toString() + " : There was a data type error: the data type of "
+        + object.getClass().getName() + " object [" + object + "] does not correspond to value meta ["
+        + toStringMeta() + "]" );
+    }
+  }
+
+  @Override
+  public Object readData( DataInputStream inputStream ) throws KettleFileException, KettleEOFException,
+    SocketTimeoutException {
+    try {
+      // Is the value NULL?
+      if ( inputStream.readBoolean() ) {
+        return null; // done
+      }
+
+      switch ( storageType ) {
+        case STORAGE_TYPE_NORMAL:
+          // Handle Content -- only when not NULL
+          long time = inputStream.readLong();
+          int nanos = inputStream.readInt();
+          Timestamp timestamp = new Timestamp( time );
+          timestamp.setNanos( nanos );
+          return timestamp;
+
+        case STORAGE_TYPE_BINARY_STRING:
+          return readBinaryString( inputStream );
+
+        case STORAGE_TYPE_INDEXED:
+          return readSmallInteger( inputStream ); // just an index: 4-bytes should be enough.
+
+        default:
+          throw new KettleFileException( toString() + " : Unknown storage type " + getStorageType() );
+      }
+    } catch ( EOFException e ) {
+      throw new KettleEOFException( e );
+    } catch ( SocketTimeoutException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)